ui/src/components/editor/shapeComponents/QueryDefinitionBuilder.vue

```diff
@@ -3,72 +3,100 @@
     <div v-if="loading" class="loading-container">
       <ProgressSpinner />
     </div>
-    <CohortEditor v-else v-model:queryDefinition="queryDefinition" />
+    <div v-else class="content-container" :class="showValidation && invalid && 'invalid'">
+      <CohortEditor v-model:queryDefinition="queryDefinition" />
+    </div>
+    <span class="error-message" v-if="validationErrorMessage"> {{ validationErrorMessage }}</span>
   </div>
 </template>
 
 <script setup lang="ts">
 import CohortEditor from "@/components/query/builder/CohortEditor.vue";
 import injectionKeys from "@/injectionKeys/injectionKeys";
-import { QueryService } from "@/services";
-import { resolveIri } from "@im-library/helpers/TTTransform";
-import { Match, Query } from "@im-library/interfaces/AutoGen";
+import { EditorMode } from "@im-library/enums";
+import { isArrayHasLength } from "@im-library/helpers/DataTypeCheckers";
+import { Match, PropertyShape, Query } from "@im-library/interfaces/AutoGen";
 import { IM } from "@im-library/vocabulary";
-import { cloneDeep } from "lodash";
 import { ComputedRef, Ref, computed, inject, onMounted, ref, watch } from "vue";
 import { useRoute } from "vue-router";
+import _ from "lodash";
```

Comment:
Since there is only one method used from lodash it's better to import only that instead of the whole package - this minimises bundle size

import { cloneDeep } from "lodash"; => import { cloneDeep } from "lodash";